### ISSUE TITLE
java.lang.VerifyError in andoid API 16 and 17 and the app crashes on launch.

### ISSUE THREAD

**Issue by saurabhtanu** [bumptech/glide#3152]
**Glide Version**: 4.0.0

**Integration libraries**: okttp 4.0.0

**Device/Android Version**: 4.1.2 and 4.2.0

**Issue details / Repro steps / Use case background**:  I have glide dependency added in the code and when I build and install the app on the above mentioned devices the app crashes on launch with the stack trace given. works fine on other android devices. My app has minSDKVerison set to 15 and compileSDKVersion set to 26

**Glide load line / `GlideModule` (if any) / list Adapter code (if any)**:
```java
public static void downloadImageAsync(final ImageView imageView, final String imageUrl) {
        Glide.with(imageView.getContext())
                .load(imageUrl)
                .into(imageView);
    }

```

**Stack trace / LogCat**:
```ruby
E/AndroidRuntime: FATAL EXCEPTION: main
                  java.lang.VerifyError: com/mine/utils/UtilsNavigation
                      at com.mine.AppLifecycleHandler.onActivityCreated(AppLifecycleHandler.java:90)
                      at android.app.Application.dispatchActivityCreated(Application.java:154)
                      at android.app.Activity.onCreate(Activity.java:887)
                      at android.support.v4.app.SupportActivity.onCreate(SupportActivity.java:66)
                      at android.support.v4.app.FragmentActivity.onCreate(FragmentActivity.java:297)
                      at android.support.v7.app.AppCompatActivity.onCreate(AppCompatActivity.java:84)
                      at com.mine.views.BaseActivity.onCreate(BaseActivity.java:32)
                      at com.mine.views.ThemeActivity.onCreate(ThemeActivity.java:36)
                      at com.mine.views.HubActivity.onCreate(HubActivity.java:162)
                      at android.app.Activity.performCreate(Activity.java:5008)
                      at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1079)
                      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2034)
                      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2095)
                      at android.app.ActivityThread.access$600(ActivityThread.java:137)
                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1206)
                      at android.os.Handler.dispatchMessage(Handler.java:99)
                      at android.os.Looper.loop(Looper.java:213)
                      at android.app.ActivityThread.main(ActivityThread.java:4786)
                      at java.lang.reflect.Method.invokeNative(Native Method)
                      at java.lang.reflect.Method.invoke(Method.java:511)
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:789)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:556)
                      at dalvik.system.NativeStart.main(Native Method)
```

**Comment by sjudd** [comment]
Are you able to reproduce with a more recent version of Glide?

**Comment by casolorz** [comment]
I have this same issue with Glide `4.11`. 

```
Fatal Exception: java.lang.VerifyError: d/c/a/n/m/m
       at com.bumptech.glide.Glide.<init>(Glide.java:428)
       at com.bumptech.glide.GlideBuilder.build(GlideBuilder.java:564)
       at com.bumptech.glide.Glide.initializeGlide(Glide.java:290)
       at com.bumptech.glide.Glide.initializeGlide(Glide.java:242)
       at com.bumptech.glide.Glide.checkAndInitializeGlide(Glide.java:201)
       at com.bumptech.glide.Glide.get(Glide.java:182)
       at com.bumptech.glide.Glide.getRetriever(Glide.java:749)
       at com.bumptech.glide.Glide.with(Glide.java:813)
       at MyFragment.doSomeMediaStuff(MyFragment.java:428)
       at MyFragment.initSomeStuff(MyFragment.java:350)
       at MyFragment.onResume(MyFragment.java:297)
       at androidx.fragment.app.Fragment.performResume(Fragment.java:2649)
       at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:922)
       at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:1238)
       at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:1303)
       at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:439)
       at androidx.fragment.app.FragmentManagerImpl.executeOps(FragmentManagerImpl.java:2079)
       at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManagerImpl.java:1869)
       at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:1824)
       at androidx.fragment.app.FragmentManagerImpl.execPendingActions(FragmentManagerImpl.java:1727)
       at androidx.fragment.app.FragmentManagerImpl$2.run(FragmentManagerImpl.java:150)
       at android.os.Handler.handleCallback(Handler.java:733)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:136)
       at android.app.ActivityThread.main(ActivityThread.java:5017)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:788)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:604)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

Any idea what the issue might be?

**Comment by TWiStErRob** [comment]
@casolorz Can you consistently reproduce this?
If yes, I would suggest trying `-dontobfuscate` in Proguard and see if it reproduces (expected to, if it was consistent before).
Then `-dontoptimize` to rule out ProGuard messing up the bytecode in .class files.

**Comment by casolorz** [comment]
I'll have to try that on my next release, for this one I went back to 4.10 and the crash reports ended.

**Comment by n00dIes** [comment]
> I'll have to try that on my next release, for this one I went back to 4.10 and the crash reports ended.

Same problem. We had to downgrade to 4.10 to prevent the app from crashing on Android 4. Please fix it.

**Comment by hiddow** [comment]
Same problem，go back 4.10 with ExifInterface: Invalid image: ExifInterface got an unsupported image format file(ExifInterface supports JPEG and some RAW image formats only) or a corrupted JPEG file to ExifInterface.

**Comment by milkaman** [comment]
I have the same problem, but updating proguard rules from this https://github.com/bumptech/glide/blob/master/library/proguard-rules.txt
solved this problem

**Comment by v1k** [comment]
Faced the same issue, but updating proguard helped.
@milkaman Thanks a lot!

**Comment by YousufSohail** [comment]
Updating proguard rules fixed the crash but images stop loading in imageview. @milkaman @v1k

**Comment by mhd-dll** [comment]
Updating proguard rules doesn't fix the crash, we had to downgrade to 4.10.0 to prevent the app from crashing
